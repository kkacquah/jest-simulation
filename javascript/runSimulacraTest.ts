const execa = require('execa');

interface JestSimulationOptions {
  args?: string[];
  [key: string]: any;
}

export async function runJestSimulation(testFile: string, options: JestSimulationOptions = {}) {
  console.log(`Running jest simulation for test file: ${testFile}`);
  const jestBin = require.resolve('jest/bin/jest');
  const { args = [], ...execaOptions } = options;
  
  const result = await execa(
    'node',
    [jestBin, testFile, '--no-color', ...args],
    {
      reject: false,
      stdio: 'inherit', 
      ...execaOptions,
    }
  );

  return result;
}