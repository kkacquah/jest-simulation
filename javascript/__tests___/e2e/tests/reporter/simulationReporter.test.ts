import { resolve } from "path";
import { runSimulacraTest } from "../../../../runSimulacraTest";

describe("SimulationReporter", () => {
  const testFile = resolve(__dirname, "../fakeTests/expectEventually.test.ts");

  it("should format successful test output correctly", async () => {
    const result = await runSimulacraTest(testFile, {
      args: [
        "-t",
        "should eventually pass when condition becomes true on third turn",
      ],
    });

    expect(result.stdout).toMatchSnapshot();
  });

  it("should format failed test output correctly", async () => {
    const result = await runSimulacraTest(testFile, {
      args: ["-t", "should fail when condition never becomes true"],
    });

    expect(result.stdout).toMatchSnapshot();
  });
});
