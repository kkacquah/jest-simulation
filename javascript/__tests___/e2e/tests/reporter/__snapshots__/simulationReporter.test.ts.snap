// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimulationReporter should format failed test output correctly 1`] = `
"
Starting Simulation Tests...
"
`;

exports[`SimulationReporter should format successful test output correctly 1`] = `
"
Starting Simulation Tests...

[32mâœ“ passing test[0m

Simulation Test Results:
Total Tests: 1
Passed: 1
Failed: 0
Duration: 0.731s
"
`;
